src/Eval.hs

```diff
@@ -293,6 +297,10 @@ macroExpand ctx xobj =
       (newCtx, expanded) <- foldlM successiveExpand (ctx, Right []) objs
       return (newCtx, do ok <- expanded
                          Right (XObj (Arr ok) i t))
+    XObj (StaticArr objs) i t -> do
+      (newCtx, expanded) <- foldlM successiveExpand (ctx, Right []) objs
+      return (newCtx, do ok <- expanded
```

Comment:
Thanks! I wonder if we should add a short comment at the beginning of this function noting this, or use the [`$!`](https://wiki.haskell.org/Performance/Strictness#Evaluating_expressions_strictly) operator instead--the extra bind might be puzzling at first glance. I actually find it to be a handy notation once you realize it's forcing, so maybe a small comment is the best way to go.

Reply:
I think that makes sense. Since this is actually a relatively common pattern in `eval` and `macroExpand`, maybe there should be a comment above `eval`? I don’t know how useful it is to move it up to where people might not see it, but it also seems annoying to duplicate it a bunch of times.

Reply:
> I think that makes sense. Since this is actually a relatively common pattern in `eval` and `macroExpand`, maybe there should be a comment above `eval`? I don’t know how useful it is to move it up to where people might not see it, but it also seems annoying to duplicate it a bunch of times.

Yep, that sounds like a good solution to me!

Reply:
I added a comment. Would you be able to take a look at it and tell me whether it makes sense?